src/Contract/UnbalancedTx.purs

```diff
@@ -35,24 +65,166 @@ import Ctl.Internal.ProcessConstraints.Error
       , TxOutRefWrongType
       , WrongRefScriptHash
+      , ValidatorHashNotFound
       , CannotSatisfyAny
       , ExpectedPlutusScriptGotNativeScript
       , CannotMintZero
       )
   ) as X
 import Ctl.Internal.ProcessConstraints.UnbalancedTx (UnbalancedTx)
 import Ctl.Internal.ProcessConstraints.UnbalancedTx (UnbalancedTx(UnbalancedTx)) as X
+import Ctl.Internal.Transaction (explainModifyTxError)
+import Ctl.Internal.Types.ByteArray (byteArrayToHex)
+import Ctl.Internal.Types.Interval (explainPosixTimeToSlotError)
+import Ctl.Internal.Types.OutputDatum
+  ( OutputDatum(NoOutputDatum, OutputDatumHash, OutputDatum)
+  )
+import Ctl.Internal.Types.PlutusData (pprintPlutusData)
 import Ctl.Internal.Types.ScriptLookups
   ( ScriptLookups
   )
 import Ctl.Internal.Types.TxConstraints (TxConstraints)
-import Data.Either (Either)
+import Data.Either (Either(Left, Right))
+import Data.FoldableWithIndex (foldMapWithIndex)
+import Data.Log.Tag as TagSet
+import Data.Maybe (Maybe(Nothing, Just))
+import Data.Newtype (unwrap)
+import Effect.Exception (error)
 
 -- | Create an `UnbalancedTx` given `ScriptLookups` and
 -- | `TxConstraints`. This should be called in conjuction with
 -- | `balanceTx` and  `signTransaction`.
-mkUnbalancedTx
+-- |
+-- | This is a 'non-throwing' variant; if you need the 'throwing' variant, use
+-- | `mkUnbalancedTx` instead.
+mkUnbalancedTxE
   :: ScriptLookups
   -> TxConstraints
   -> Contract (Either MkUnbalancedTxError UnbalancedTx)
-mkUnbalancedTx = PC.mkUnbalancedTxImpl
+mkUnbalancedTxE = PC.mkUnbalancedTxImpl
+
+-- | As `mkUnbalancedTxE`, but 'throwing'.
+mkUnbalancedTx :: ScriptLookups -> TxConstraints -> Contract UnbalancedTx
+mkUnbalancedTx lookups constraints =
+  mkUnbalancedTxE lookups constraints >>= case _ of
+    Left err -> throwError $ error $ explainMkUnbalancedTxError err
+    Right res -> pure res
+
+-- | Helper to pretty-print `MkUnbalancedTxError`s.
+explainMkUnbalancedTxError :: MkUnbalancedTxError -> String
+explainMkUnbalancedTxError = case _ of
+  CannotFindDatum -> "Cannot find datum"
+  CannotQueryDatum dh ->
+    "Querying for datum by datum hash ("
+      <> show dh
+      <>
+        ") failed: no datum found"
+  CannotConvertPOSIXTimeRange tr ttsErr ->
+    "Cannot convert POSIX time range to slot time range.\nRange: "
+      <> show tr
+      <> "\nReason: "
+      <>
+        explainPosixTimeToSlotError ttsErr
+  CannotSolveTimeConstraints tr tr' ->
+    "Unsolvable time constraints: " <> show tr <> " and " <> show tr' <>
+      " do not overlap."
+  CannotGetMintingPolicyScriptIndex ->
+    "Cannot get minting policy script index. This should be impossible.\n"
+      <> "Please report this as a bug here: "
+      <> bugTrackerLink
+  CannotGetValidatorHashFromAddress addr ->
+    "Cannot get a payment validator hash from address " <> show addr
+  CannotMakeValue _ tn _ ->
+    "Attempted to make an amount with the ADA currency symbol, and non-empty token name "
+      <> show tn
```

Comment:
Let's use the same method for showing token names as in pprintValue: text if decodable, hex if binary.